### PR TITLE
Remove VPU_MYRIAD_PLATFORM from compile tools

### PR DESCRIPTION
**Pull request by DariaMityagina**
it seems that the part where the mentioned option is used is not needed.

`IE_SUPPRESS_DEPRECATED_START`
`        config[VPU_MYRIAD_CONFIG_KEY(PLATFORM)] = "VPU_MYRIAD_2480";`
`IE_SUPPRESS_DEPRECATED_END`

Because it was there to set platform = myriad, which is actually redundant since in both compile_tool and myriad_compile device type is defined using 

`ie.LoadNetwork(network, FLAGS_d, configure());` where FLAGS_d = "MYRIAD" or
`ie.LoadNetwork(network, "MYRIAD", configure(FLAGS_c, FLAGS_m));`

Without this option model is compiled as well.

Ticket: 48398

### CODE REVIEW

**Comment by DariaMityagina** [comment]
@AntonDudchenko could you please take a look?

**Comment by AntonDudchenko** [comment]
LGTM, but please correct the mention of VPU_MYRIAD_2450 key also here:
https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_DG/supported_plugins/MYRIAD.md

**Comment by DariaMityagina** [comment]
> LGTM, but please correct the mention of VPU_MYRIAD_2450 key also here:
> https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_DG/supported_plugins/MYRIAD.md

done

**Comment by DariaMityagina** [comment]
@AntonDudchenko do you have other comments?

**Review by AntonDudchenko — APPROVED** [review]

**Comment by DariaMityagina** [comment]
@Maxim-Doronin could you please take a look?

**Review by KorA6 — APPROVED** [review]

**Comment by DariaMityagina** [comment]
@Maxim-Doronin can this PR be merged?

File: inference-engine/include/vpu/myriad_plugin_config.hpp
Comment (Maxim-Doronin) [inline inference-engine/include/vpu/myriad_plugin_config.hpp:57]: Why has it remained? Do we need `DECLARE_VPU_MYRIAD_CONFIG_KEY(PLATFORM);` config if it can take only one value?

Comment (DariaMityagina) [reply inference-engine/include/vpu/myriad_plugin_config.hpp:57]: Right, will be removed. Thanks

File: inference-engine/src/vpu/graph_transformer/src/model/stage.cpp
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/model/stage.cpp:200]: Have to remove this comment as well

File: inference-engine/src/vpu/myriad_plugin/myriad_config.cpp
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_config.cpp:22]: just return `ParsedConfig::getCompileOptions()`?

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp:272]: Let's tell a user that myriad plugin is going to compile model but inference will be unavailable

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp:379]: just `2 * executors`

**Review by Maxim-Doronin — COMMENTED** [review]
@AntonDudchenko @KorA6, please follow up with my comments and make a review once again

**Comment by DariaMityagina** [comment]
@AntonDudchenko @KorA6, please make a review once again?

**Review by AntonDudchenko — APPROVED** [review]

**Comment by DariaMityagina** [comment]
@Maxim-Doronin can it be merged now?

**Comment by AntonDudchenko** [comment]
@Maxim-Doronin please take a look, we can merge it if you have no more comments

**Review by KorA6 — APPROVED** [review]

File: inference-engine/src/vpu/myriad_plugin/myriad_config.cpp
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_config.cpp:19]: small nitpick: if ParsedConfig is a parent class then we do not need an overridden method in MyriadConfig class

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp:271]: Do we need https://github.com/openvinotoolkit/openvino/blob/bee21b5c5b3c95d05097df9ccfd95e109b1215e6/inference-engine/src/vpu/myriad_plugin/myriad_config.h#L55 method? 
It has NC_ANY_PLATFORM value by default but is never changed
So it looks for me that this condition is always true

Comment (Maxim-Doronin) [reply inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp:271]: I mean, we can possibly remove 
https://github.com/openvinotoolkit/openvino/blob/bee21b5c5b3c95d05097df9ccfd95e109b1215e6/inference-engine/src/vpu/myriad_plugin/myriad_config.h#L90
and  
https://github.com/openvinotoolkit/openvino/blob/bee21b5c5b3c95d05097df9ccfd95e109b1215e6/inference-engine/src/vpu/myriad_plugin/myriad_config.h#L55
and then don't use 
`if (config.platform() == NC_ANY_PLATFORM) {` 
condition

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.h
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_executor.h:67]: The same here

File: inference-engine/thirdparty/movidius/mvnc/include/mvnc.h
Comment (Maxim-Doronin) [inline inference-engine/thirdparty/movidius/mvnc/include/mvnc.h:130]: The same here

**Review by Maxim-Doronin — COMMENTED** [review]
Please, consider the removal enumeration ANY_PLATFORM and MYRIAD_X as it has no difference now
NB: place applicable exceptions if firmware or booted device return something that is not equal to 2x8x (myriadx)

**Comment by DariaMityagina** [comment]
product_configs check is in progress

**Comment by DariaMityagina** [comment]
@gladilov-gleb could you please take a look? Need one final review to be sure of the changes

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp
Comment (DariaMityagina) [reply inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp]: done

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.h
Comment (DariaMityagina) [reply inference-engine/src/vpu/myriad_plugin/myriad_executor.h]: done

**Comment by gladilov-gleb** [comment]
> @gladilov-gleb could you please take a look? Need one final review to be sure of the changes. Thanks

I think it's better to get working (CI is red now) version first. If you think current version is production-ready or you need help - please contact me

**Comment by gladilov-gleb** [comment]
@Maxim-Doronin @DariaMityagina 

Should this request be rebase upon https://github.com/openvinotoolkit/openvino/pull/6176 ?

**Comment by DariaMityagina** [comment]
> > @gladilov-gleb could you please take a look? Need one final review to be sure of the changes. Thanks
> 
> I think it's better to get working (CI is red now) version first. If you think current version is production-ready or you need help - please contact me

Yes, right, I am working on it

**Comment by DariaMityagina** [comment]
> @Maxim-Doronin @DariaMityagina
> 
> Should this request be rebase upon #6176 ?

I did a rebase after the mentioned PR merge.

**Comment by DariaMityagina** [comment]
@Maxim-Doronin @gladilov-gleb product-configs checks pass. All of the links are in the ticket.

**Review by Maxim-Doronin — APPROVED** [review]

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp
Comment (gladilov-gleb) [inline inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp:129]: What about compiling with no device? Is it still supported that if there is no device compilation still successes, but an attempt to create an executable network will throw?

**Review by gladilov-gleb — APPROVED** [review]

File: inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp
Comment (DariaMityagina) [reply inference-engine/src/vpu/myriad_plugin/myriad_executor.cpp:129]: In case when there is no device compilation successes. And no, there is no throw .

**Comment by DariaMityagina** [comment]
https://dsp-ci-icv.inn.intel.com/job/MYRIAD/job/IE-MDK/job/manual/job/RunTests-MYRIAD/2404/
seems to be okay
product-config check is in progress

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/28

**Pull request merged**
